packages/core/src/controllers/SendController.ts

```diff
@@ -154,6 +160,54 @@ export const SendController = {
     }
   },
 
+  async fetchNetworkBalance() {
+    const balances = await SwapApiUtil.getMyTokensWithBalance()
+
+    if (!balances) {
+      return
+    }
+
+    const networkToken = balances.find(
+      token => token.address === ChainController.getActiveNetworkTokenAddress()
+    )
+
+    if (!networkToken) {
+      return
+    }
+
+    state.networkBalanceInUSD = networkToken
+      ? NumberUtil.multiply(networkToken.quantity.numeric, networkToken.price).toString()
+      : '0'
+  },
+
+  isInsufficientNetworkTokenForGas(networkBalanceInUSD: string, gasPriceInUSD: number | undefined) {
+    const gasPrice = gasPriceInUSD || '0'
+
+    if (NumberUtil.bigNumber(networkBalanceInUSD).isZero()) {
+      return true
+    }
+
+    return NumberUtil.bigNumber(NumberUtil.bigNumber(gasPrice)).isGreaterThan(networkBalanceInUSD)
+  },
+
+  hasInsufficientToken() {
```

Comment:
Is this insufficient gas or unsupported token ?

Reply:
Gas funds, will change the name